imenu-list.el

```diff
@@ -539,7 +539,7 @@ imenu entries did not change since the last update."
       (unless (and (null force-update)
                    imenu-list--last-location
                    (marker-buffer imenu-list--last-location)
```

Comment:
I'm not sure the intention here, but if once imenu-list--last-location becomes non buffer location once, imenu-list refuses future updates? (I never experienced such case though)
We maybe should check `(marker-buffer location)` is nil, instead?